File: src/main/java/de/digitalemil/eagle/Screen.java
package de.digitalemil.eagle;

import de.digitalemil.tocplusplus.MethodDefinitionChangerAnnotation;

public class Screen {
	protected int screen;
	public final static int STAYONSCREEN= -1;
	public final static int MAXSCREEN= 256;
	protected static Screen[] screens;//
	protected static int activeScreen= -1;
	private long _start, frames;
	protected int fps, id, numberOfThings;
	protected static int nextscreen= STAYONSCREEN;
	
	public Screen() {
		_start = 0;
	}
	
	@MethodDefinitionChangerAnnotation({ "BY", "Thing[]", "Thing**", "BY", "things[i]=null", "delete things[i]", "BY", "things=null", "delete [ ] things" })
	protected void finalize() throws Throwable {
		Thing[] things= Globals.getAllThings();
		
		for(int i= 0; i< numberOfThings; i++) {
			things[i]= null;
		}
	}
	
	public int getNumberOfThings() {
		return numberOfThings;
	}
	
	public int getFps() {
		return fps;
	}
	
	public void update(long currentTimeMillis) {
	//	System.out.println("update: "+ frames);
		Globals.frames++;
		frames++;
		if (_start == 0)
			_start = PartAnimation.currentTimeMillis();

		long now = PartAnimation.currentTimeMillis();
		if (now - _start > 1000) {
			fps = (int)((1000 * frames) / (now - _start));
			_start = 0;
			frames = 0;
		}
	}

	public static Screen getActiveScreen() {
		if(activeScreen< 0 || activeScreen>= MAXSCREEN)
			return null;
		return screens[activeScreen];
	}
		
	
	public void activate() {
		activeScreen= getScreenID();
		System.out.println("Screen "+ this);
		System.out.println("Screen "+ getActiveScreen());
		nextscreen= STAYONSCREEN;	
	}
	
	@MethodDefinitionChangerAnnotation({ "BY", "int i", "int i; finalize()" })
	public void deactivate() {
		int i;
	}
	
	static public int moveToOtherScreen() {
		return nextscreen;
	}
	
	public int fillThingArray(Thing[] things) {
		
		return getNumberOfThings();
	}
	
	public void touch(int x, int y) {
	}

	public boolean touchStart(int x, int y) {
		return false;
	}

	public boolean touchStop(int x, int y) {
		return false;
	}
	
	public int getScreenID() {
		return id;
	}
	
	public static Screen getScreen(int id) {
		return screens[id];
	}
	
	public static void registerScreen(Screen screen) {
		screens[screen.getScreenID()]= screen;
	}

	public int getBackgroundColor() {
		return 0;
	}
	
}

File: platforms/ios/Tatanka/Tatanka/eagletatanka/src/screen.cpp
#include "all.h"

#include "screen.h"

Screen::~Screen() {
	finalize();

}

Screen::Screen() {
	screen = 0;
	_start = 0;
	frames = 0;
	fps = 0;
	id = 0;
	numberOfThings = 0;

  _start=0;
}


void Screen::finalize() {

  Thing** things=Globals::getAllThings();
  for (int i=0; i < numberOfThings; i++) {
    delete things[i];
  }
}


int Screen::getNumberOfThings() {

  return numberOfThings;
}


int Screen::getFps() {

  return fps;
}


void Screen::update(long currentTimeMillis) {

  Globals::frames++;
  frames++;
  if (_start == 0)   _start=PartAnimation::currentTimeMillis();
  long now=PartAnimation::currentTimeMillis();
  if (now - _start > 1000) {
    fps=(int)((1000 * frames) / (now - _start));
    _start=0;
    frames=0;
  }
}


Screen* Screen::getActiveScreen() {

  if (activeScreen < 0 || activeScreen >= MAXSCREEN)   return 0;
  return screens[activeScreen];
}


void Screen::activate() {

  activeScreen=getScreenID();
  //System::out->println((unsigned char*)"Screen " + this);
  //System::out->println((unsigned char*)"Screen " + getActiveScreen());
  nextscreen=STAYONSCREEN;
}


void Screen::deactivate() {

  int i; finalize();
}


int Screen::moveToOtherScreen() {

  return nextscreen;
}


int Screen::fillThingArray(Thing** things) {

  return getNumberOfThings();
}


void Screen::touch(int x, int y) {

}


bool Screen::touchStart(int x, int y) {

  return false;
}


bool Screen::touchStop(int x, int y) {

  return false;
}


int Screen::getScreenID() {

  return id;
}


Screen* Screen::getScreen(int id) {

  return screens[id];
}


void Screen::registerScreen(Screen* screen) {

  screens[screen->getScreenID()]=screen;
}


int Screen::getBackgroundColor() {

  return 0;
}


Screen** Screen::screens = null;
int Screen::activeScreen = -1;
int Screen::nextscreen = STAYONSCREEN;
